LEGO1/mxbitmap.cpp

```diff
@@ -0,0 +1,105 @@
+#include "mxbitmap.h"
+#include "mxbool.h"
+#include "mxpalette.h"
+
+// OFFSET: LEGO1 0x100bc980
+MxBitmap::MxBitmap()
+{
+    this->m_info = NULL;
+    this->m_bmiHeader = NULL;
+    this->m_paletteData = NULL;
+    this->m_data = NULL;
+    this->m_unk18 = 0;
+    this->m_palette = NULL;
+}
+
+// OFFSET: LEGO1 0x100bca10
+MxBitmap::~MxBitmap()
+{
+    if (this->m_info != NULL) {
+        free(this->m_info);
+    }
+    if (this->m_data != NULL) {
+        free(this->m_data);
+    }
+    if (this->m_palette != NULL) {
+        // FIXME: Once m_palette/base is understood
+    }
+    // FIXME: Unwind@100bca91. Stack unwinding, I've never done this before.
+}
+
+// OFFSET: LEGO1 0x100bd1c0
+MxPalette *MxBitmap::CreatePalette()
+{
+    // FIXME: This function needs MxPalette to be completed. Also INFERRING usage of MxBool
+    MxPalette *pal;
+    MxPalette *ppal;
+    MxBool success;
+
+    pal = NULL;
+    success = MX_FALSE;
+    if(this->m_unk18 == '\0') {
+        ppal = (MxPalette *)malloc(0x414);
+        pal = NULL;
+        if(ppal != NULL) {
+            // pal = MxPalette::FromBitmapPalette(ppal, this->m_paletteData);
+        }
```

Comment:
This is probably using an alternative c++ constructor, or calling `MxPalette::FromBItmapPalette` directly.
It doesn't make sense in c++ to allocate, and then call a method on raw data.
```suggestion
        // ppal = MxPalette::FromBitmapPalette(this->m_paletteData);
```

Reply:
@madebr is correct, 0x100BEED0 is MxPalette ctor taking a const RGBQUAD* (or const PALETTEENTRY*).

Additionally, what's spelled `malloc` here isn't malloc, it's the global operator new function. I.e. what's here spelled out as `malloc(0x414);` can in step 1 be changed to `malloc(sizeof MxPalette);` (providing a chance to verify with a compile-time assert that the declaration of MxBitmap is indeed 0x414), and for its final step it becomes
`pal = new MxPalette(m_paletteData);`. The null-check is something the Microsoft compilers insert, since their allocators by default do not throw on out-of-memory, but simply return 0.